src/prefect/task_worker.py

```diff
@@ -156,12 +156,24 @@ async def _subscribe_to_task_scheduling(self):
             client_id=self._client_id,
             base_url=base_url,
         ):
+            logger.info(f"Received task run: {task_run.id} - {task_run.name}")
             if self._limiter:
                 await self._limiter.acquire_on_behalf_of(task_run.id)
-            logger.info(f"Received task run: {task_run.id} - {task_run.name}")
             self._runs_task_group.start_soon(self._submit_scheduled_task_run, task_run)
 
     async def _submit_scheduled_task_run(self, task_run: TaskRun):
+        try:
+            await self._submit_and_start_run(task_run)
+        except BaseException as exc:
+            logger.exception(
+                f"Failed to submit task run {task_run.id!r}",
+                exc_info=exc,
+            )
+        finally:
+            if self._limiter:
+                self._limiter.release_on_behalf_of(task_run.id)
+
+    async def _submit_and_start_run(self, task_run: TaskRun):
```

Comment:
name suggestions appreciated for this or the parent function, I didn't want to inline everything in a giant try/except

Reply:
Maybe `_safe_submit_scheduled_task_run` and `_submit_scheduled_task_run`?